Objective-C/CBLDocument.mm

```diff
@@ -330,52 +397,61 @@ - (BOOL) mergeWithConflictResolver: (id<CBLConflictResolver>)resolver
                           deletion: (bool)deletion
                              error: (NSError**)outError
 {
-    // Read the current revision from the database, and parse it into an NSDictionary:
-    C4Document *currentDoc = [self readC4Doc_mustExist: YES error: outError];
-    if (!currentDoc)
+    // Read the current revision from the database:
+    C4Document* rawDoc = [self readC4Doc_mustExist: YES error: outError];
+    if (!rawDoc)
         return NO;
-    NSDictionary *current = nil;
-    auto currentData = currentDoc->selectedRev.body;
-    if (currentData.size > 0) {
-        FLValue currentRoot = FLValue_FromTrustedData({currentData.buf, currentData.size});
-        cbl::SharedKeys currentKeys(*self.sharedKeys, (FLDict)currentRoot);
-        current = FLValue_GetNSObject(currentRoot, &currentKeys);
-    }
-
-    NSDictionary* resolved;
+    
+    FLDict curRoot = nullptr;
+    auto curBody = rawDoc->selectedRev.body;
+    if (curBody.size > 0)
+        curRoot = (FLDict) FLValue_FromTrustedData({curBody.buf, curBody.size});
+    
+    // Create the current readonly document with the current revision:
+    CBLC4Document* curC4doc = [CBLC4Document document: rawDoc];
+    CBLFLDict* curDict = [[CBLFLDict alloc] initWithDict: curRoot
```

Comment:
As far as there is no circular reference,  it should get released? Do you see anything that could prevent the release to happen?